kore/src/Kore/Syntax/Pattern.hs

```diff
@@ -238,13 +260,178 @@ instance TopBottom (Pattern variable annotation) where
     isBottom (Recursive.project -> _ :< BottomF _) = True
     isBottom _ = False
 
-instance
-    (Unparse variable)
-    => SQL.Column (Pattern variable annotation)
-  where
+instance Unparse variable => SQL.Column (Pattern variable annotation) where
     defineColumn tableName _ = SQL.defineColumn tableName (SQL.Proxy @Text)
     toColumn = SQL.toColumn . Pretty.renderText . Pretty.layoutOneLine . unparse
 
+instance
+    From
+        (And Sort (Pattern variable Attribute.Null))
+        (Pattern variable Attribute.Null)
+  where
+    from = Recursive.embed . (:<) Attribute.Null . from
+    {-# INLINE CONLIKE from #-}
```

Comment:
I don't think I fully understand what `CONLIKE` does from the GHC documentation, I'll keep in mind to try to learn more about it, but could you give me some hints related to why it was necessary here? Did something not work properly or is there something specific about this code which makes a pragma like this necessary?

Reply:
`CONLIKE` is short for "constructor-like". This tells GHC that this function is inexpensive to compute, like a constructor. This will make the compiler very aggressive about inlining it. I'm using it here because I know that all these functions put together are just applying several constructors on top of the argument.

Reply:
I understand now, thank you!